tests/system_tests_split_path.py

```diff
@@ -0,0 +1,235 @@
+
+    def on_io_event(self, bsock, mask, peer_bsock):
+        # If writable send any buffered data
+        if mask & selectors.EVENT_WRITE == selectors.EVENT_WRITE:
+            sent = bsock.sock.send(bsock.input_buf)
+            self._logger.log(f"MyLoadBalancer sending {sent} bytes to {bsock.sock.getpeername()}")
+            del bsock.input_buf[:sent]
+            if not bsock.input_buf:
+                # No more to write
+                self.selector.modify(bsock, selectors.EVENT_READ, peer_bsock)
+
+        # If readable fill the input buffer of the peer
+        if mask & selectors.EVENT_READ == selectors.EVENT_READ:
+            data = bsock.sock.recv(4096)
+            if data:
+                self._logger.log(f"MyLoadBalancer receiving {len(data)} bytes from {bsock.sock.getpeername()}")
+                peer_bsock.input_buf += data
+                self.selector.modify(peer_bsock,
+                                     selectors.EVENT_READ | selectors.EVENT_WRITE,
+                                     bsock)
+            else:
+                self._logger.log(f"MyLoadBalancer closing conns {bsock.sock.getpeername()} {peer_bsock.sock.getpeername()}")
+                self.selector.unregister(bsock)
+                bsock.sock.shutdown(socket.SHUT_RDWR)
+                bsock.sock.close()
+                self._socks.remove(bsock)
+
+                self.selector.unregister(peer_bsock)
+                peer_bsock.sock.shutdown(socket.SHUT_RDWR)
+                peer_bsock.sock.close()
+                self._socks.remove(peer_bsock)
+
+    def teardown(self):
+        self._shutdown = True
+        self._thread.join(TIMEOUT)
+        if self._thread.is_alive():
+            raise Exception("MyLoadBalancer failed to join!")
+
+
+class InterRouterSplitPathTest(TestCase):
+    """
+    Verify the router logs errors when it detects that connections from the
+    same connection group arrive a different routers
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(InterRouterSplitPathTest, cls).setUpClass()
+
+    def router(self, name, test_config, data_connection_count, **kwargs):
+        config = [
+            ('router', {'mode': 'interior',
+                        'id': name,
+                        'dataConnectionCount': f"{data_connection_count}"}),
+            ('listener', {'port': self.tester.get_port(), 'role': 'normal'}),
+        ]
+        config.extend(test_config)
+        return self.tester.qdrouterd(name, Qdrouterd.Config(config), **kwargs)
+
+    def test_01(self):
+        """
+        Simulate splitting connections to the wrong routers.  Creates a single
+        connector-based router and two listener-based routers. A mock load
+        balancer sits between the connector router and the listeners. The mock
+        load balancer splits the connection group connections between the two
+        listener routers
+        """
+        data_conn_count = 8
```

Comment:
In production, the data_conn_count usually defaults to 2, can we leave it at 2 to mimic production use case ?

Reply:
Done.